### PR TITLE
suspend anonymously

### LINKED ISSUE #2614: [Suspend] Suspension notification shouldn't reveal the actor
## Feature Request

**Is your feature request related to a problem? Please describe.**

As discussed* [on Discuss](https://discuss.flarum.org/d/26161-anonymous-suspension), the notification of suspension of a user reveals who was the moderator/admin that took the action, although it is unnecessary and could unfortunately be risky for the actor user.

*It wasn't actually discussed but I'll take the likes as an approval to open this issue.

**Describe the solution you'd like**

The suspension notification shouldn't include the name of the user that took the action, which is easy to change:

https://github.com/flarum/suspend/blob/b98e31963f6e82803599b56c157b9de8b266c959/locale/en.yml#L17-L19

But it should also not have a profile picture shown. I'm not sure if this is technically possible with how notifications currently work in Flarum, because after taking a quick look it appears that the actor is required in the notification blueprint.

### PR DESCRIPTION
**Pull request by hasan-ozbey**
Closes https://github.com/flarum/core/issues/2614

I've noticed this issue while I'm scrolling through the core issues. I'm already using this type of feature in my forum. It's up to you if it's worth to merge.

### CODE REVIEW

**Comment by tankerkiller125** [comment]
Thanks for the contribution to Flarum, looking at the code it appears that this PR completely replaces the non-anonymous function of suspend, given the change I'm not going to approve this or reject the changes until the team can decide if this is the behavior we want, or if we would want this to be an optional change via settings.

Again, thanks for the PR

File: src/Notification/UserUnsuspendedBlueprint.php [hunk 1]
```diff
@@ -47,7 +40,6 @@ public function getSubject()
      */
     public function getFromUser()
     {
-        return $this->actor;
     }
```
Comment (luceos) [inline src/Notification/UserUnsuspendedBlueprint.php]: Does a reason exist for `getFromUser` to remain in this class? Is it a requirement of an interface by any chance? If so, it should probably return `null` or be removed entirely.

Same applies for the other Blueprint.

**Review by luceos — CHANGES_REQUESTED** [review]
I'm overall okay with this. At first I considered that we might need to keep the user intact on the Blueprints. But we already have events in case logic needs to be extended.

I left one remark though that needs a change.

File: src/Notification/UserUnsuspendedBlueprint.php (hunk 1 above)
Comment (hasan-ozbey) [reply src/Notification/UserUnsuspendedBlueprint.php]: yes, it's here https://github.com/flarum/core/blob/eb4b18a979c7406cbf154a107662652d282fe415/src/Notification/Blueprint/BlueprintInterface.php#L22-L27 I left it this way to keep up with `getData` (also returns nothing and yet it's in the class) https://github.com/flarum/suspend/blob/245fc135a233a9a13d48c7bbcd3941fb4c626968/src/Notification/UserUnsuspendedBlueprint.php#L53-L58

**Review by davwheat — APPROVED** [review]
I think this is a good idea. It might be worth showing who suspended the user to other admins, via the suspension interface, but that's down to another PR.

Moderation actions such as this should be performed as a group, rather than individual users. Displaying the user only causes more trouble, in my opinion.

File: src/Notification/UserUnsuspendedBlueprint.php (hunk 1 above)
Comment (luceos) [reply src/Notification/UserUnsuspendedBlueprint.php]: Thanks for adding those. It seems the interface actually does require null or the return value, but don't have proper return type hinting yet 😞

**Review by luceos — APPROVED** [review]

**Pull request merged**